### PR TITLE
fix-py37-update-image-version

### PR DESCRIPTION
**Pull request by dendi**
fix(py37): update some dependencies and image version to use python-37-slim

### CODE REVIEW

File: py37-slim/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  && pip install \
+    Cython==0.27.3 \
+    cymem==1.31.2 \
+    preshed==1.0.1 \
+    thinc==6.10.3 \
+    murmurhash==0.28.0 \
+    ujson==1.35 \
+    regex==2017.4.5 \
+  && pip install git+https://github.com/KurioApp/spaCy.git@8aa806d4c6031d7b6929edb01f86be41b89aae10#egg=spacy
+RUN apt-get update && \
```
Comment (ekapratama93) [inline py37-slim/Dockerfile]: Don't need update

File: py37-slim/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM python:3.7-slim
+RUN apt-get update \
+&& apt-get upgrade -y \
+&& apt install build-essential -y \
+&& apt-get install git -y \
+&& apt-get clean
```
Comment (ekapratama93) [inline py37-slim/Dockerfile]: Move to last

File: py37-slim/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM python:3.7-slim
+RUN apt-get update \
+&& apt-get upgrade -y \
+&& apt install build-essential -y \
+&& apt-get install git -y \
```
Comment (ekapratama93) [inline py37-slim/Dockerfile]: Can combined with build-essential

File: py37-slim/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
+FROM python:3.7-slim
+RUN apt-get update \
+&& apt-get upgrade -y \
+&& apt install build-essential -y \
```
Comment (ekapratama93) [inline py37-slim/Dockerfile]: APT or apt-get?

**Review by ekapratama93 — CHANGES_REQUESTED** [review]

File: py37-slim/Dockerfile (hunk 2 above)
Comment (williamgun007) [inline py37-slim/Dockerfile]: add space (we use 2 space)

File: py37-slim/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,21 @@
+FROM python:3.7-slim
+RUN apt-get update \
```
Comment (williamgun007) [inline py37-slim/Dockerfile:3]: add blank line between the execution

File: py37-slim/Dockerfile (hunk 1 above)
Comment (williamgun007) [inline py37-slim/Dockerfile]: add blank line between the execution

File: py37-slim/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    Cython==0.27.3 \
+    cymem==1.31.2 \
+    preshed==1.0.1 \
+    thinc==6.10.3 \
+    murmurhash==0.28.0 \
+    ujson==1.35 \
+    regex==2017.4.5 \
+  && pip install git+https://github.com/KurioApp/spaCy.git@8aa806d4c6031d7b6929edb01f86be41b89aae10#egg=spacy
+RUN apt-get update && \
+apt-get remove --auto-remove build-essential -y
```
Comment (williamgun007) [inline py37-slim/Dockerfile]: add space (we use 2 space)

**Review by williamgun007 — CHANGES_REQUESTED** [review]

File: py37-slim/Dockerfile (hunk 3 above)
Comment (dendi) [reply py37-slim/Dockerfile]: Git does not include in the build-essential package
https://packages.ubuntu.com/xenial/build-essential

Comment (ekapratama93) [reply py37-slim/Dockerfile]: I mean `apt-get install -y build-essential git`

File: py37-slim/Dockerfile (hunk 2 above)
Comment (dendi) [reply py37-slim/Dockerfile]: I think `apt-get install -y build-essential git` must be executed first before `pip install`
I got error when executing via my local env

Comment (ekapratama93) [reply py37-slim/Dockerfile]: move only clean up part

**Comment by ekapratama93** [comment]
Use Dockerfile linter if yo need. https://hadolint.github.io/hadolint/

File: py37-slim/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,24 @@
+FROM python:3.7-slim
+
+RUN apt-get update \
+&& apt-get upgrade -y \
+&& apt-get install -y --no-install-recommends build-essential git \
+&& rm -rf /var/lib/apt/lists/*
```
Comment (ekapratama93) [inline py37-slim/Dockerfile]: add double spaces

File: py37-slim/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    preshed==1.0.1 \
+    thinc==6.10.3 \
+    murmurhash==0.28.0 \
+    ujson==1.35 \
+    regex==2017.4.5 \
+  && pip install git+https://github.com/KurioApp/spaCy.git@8aa806d4c6031d7b6929edb01f86be41b89aae10#egg=spacy
+
+RUN apt-get clean \
+&& rm -rf /var/lib/apt/lists/* \
+&& apt-get remove --auto-remove build-essential -y
```
Comment (ekapratama93) [inline py37-slim/Dockerfile]: add double space

**Review by ekapratama93 — CHANGES_REQUESTED** [review]

File: py37-slim/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+RUN pip install -U pipenv \
+  && export PATH=$(realpath ~)/.local/bin:$PATH \
+  && pip install \
+     Cython==0.27.3 \
+    cymem==1.31.2 \
+    preshed==1.0.1 \
+    thinc==6.10.3 \
+    murmurhash==0.28.0 \
+    ujson==1.35 \
+    regex==2017.4.5 \
```
Comment (ekapratama93) [inline py37-slim/Dockerfile]: indentation

**Review by ekapratama93 — CHANGES_REQUESTED** [review]

**Review by ekapratama93 — APPROVED** [review]
LGTM

**Review by williamgun007 — APPROVED** [review]
LGTM

**Pull request merged**